src/main/java/com/netchill/db/dao/movie/GenreDao.java

```diff
@@ -0,0 +1,38 @@
+package com.netchill.db.dao.movie;
+
+import com.coreoz.plume.db.querydsl.transaction.TransactionManagerQuerydsl;
+import com.netchill.db.generated.Genre;
+import com.netchill.db.generated.QGenre;
+import com.netchill.db.generated.QPreviewGenreMovie;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.List;
+
+@Singleton
+public class GenreDao {
+    private final TransactionManagerQuerydsl transactionManager;
+    @Inject
+    public GenreDao(TransactionManagerQuerydsl transactionManager){
+        this.transactionManager = transactionManager;
+    }
+
+    public Genre getGenreById(Long id) {
+        return this.transactionManager.selectQuery()
+                .select(QGenre.genre)
+                .from(QGenre.genre)
+                .where(QGenre.genre.id.eq(id))
+                .fetchOne();
```

Comment:
tu as 2 méthode pour récupérer un seul élément:
- `fetchFirst`: récupère le premier élément et le renvoie sous forme d'Optionel. L'Optional est vide si aucun élément trouvé. 
- `fetchOne`: récupère un seul élément, si il y en as plusieurs renvoie une exception et return null si aucun élément présent.

Ici tu as bien fait d'utiliser `fetchOne` car ce n'est pas normal d'avoir plusieurs genre avec le même id, donc si ça arrive il faut lever une exception.

Cependant la valeur peut être null, pour que les personnes qui utilisent ta méthode le sache renvoie un Optionel.

```suggestion
        return Optional.ofNullable(this.transactionManager.selectQuery()
                .select(QGenre.genre)
                .from(QGenre.genre)
                .where(QGenre.genre.id.eq(id))
                .fetchOne());
```